### PR TITLE
[BW-939] Pass GSA to Cromwell helm chart to run compute as pet account

### PR DESCRIPTION
**Pull request by salonishah11**
Edit: Cromwhelm PR has now been merged! ~**Note: This should merge after cromwhelm PR: https://github.com/broadinstitute/cromwhelm/pull/18. It is currently a draft because without the cromwhelm changes, the tests will fail.**~

Closes https://broadworkbench.atlassian.net/browse/BW-939

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreterSpec.scala [hunk 1]
```diff
@@ -155,10 +155,11 @@ class GKEInterpreterSpec extends AnyFlatSpecLike with TestComponent with Leonard
...
+    res.mkString(",") shouldBe """nodeSelector.cloud\.google\.com/gke-nodepool=pool1,persistence.size=250G,persistence.gcePersistentDisk=disk1,env.swaggerBasePath=/proxy/google/v1/apps/dsp-leo-test1/app1/cromwell-service,config.gcsProject=dsp-leo-test1,config.gcsBucket=gs://test-bucket/cromwell-execution,config.serviceAccount.name=app1-cromwell-ksa,config.serviceAccount.annotations.gcpServiceAccount=pet123-abc@terra-test-abc.iam.gserviceaccount.com,ingress.enabled=true,ingress.annotations.nginx\.ingress\.kubernetes\.io/proxy-redirect-from=https://1455694897.jupyter.firecloud.org,ingress.annotations.nginx\.ingress\.kubernetes\.io/proxy-redirect-to=https://leo,ingress.annotations.nginx\.ingress\.kubernetes\.io/rewrite-target=/$2,ingress.annotations.nginx\.ingress\.kubernetes\.io/auth-tls-secret=ns/ca-secret,ingress.path=/proxy/google/v1/apps/dsp-leo-test1/app1/cromwell-service,ingress.hosts[0].host=1455694897.jupyter.firecloud.org,ingress.hosts[0].paths[0]=/proxy/google/v1/apps/dsp-leo-test1/app1/cromwell-service(/|$)(.*),ingress.tls[0].secretName=tls-secret,ingress.tls[0].hosts[0]=1455694897.jupyter.firecloud.org,db.password=replace-me"""
```
Comment (cjllanwarne) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreterSpec.scala]: TOL this is a little hard to read (and thus easy to make mistakes)... I wonder whether splitting across many lines would help, eg
```
shouldBe """nodeSelector.cloud\.google\.com/gke-nodepool=pool1,""" + 
"""persistence.size=250G,""" +
"""persistence.gcePersistentDisk=disk1,""" + 
...
```

**Review by cjllanwarne — APPROVED** [review]

**Review by kpierre13 — APPROVED** [review]

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreterSpec.scala (hunk 1 above)
Comment (kpierre13) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreterSpec.scala]: I +1 the idea of a multiline comment for readability. But, I'm also a biased newbie.

Comment (salonishah11) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreterSpec.scala]: I have updated it to be multi-line 👍

**Review by Qi77Qi — APPROVED** [review]
once tests pass 👍

**Comment by salonishah11** [comment]
jenkins retest

**Comment by salonishah11** [comment]
jenkins retest

**Comment by salonishah11** [comment]
jenkins retest

**Pull request merged**